x-pack/legacy/plugins/infra/public/components/logging/log_text_stream/column_headers.tsx

```diff
@@ -30,13 +33,16 @@ export const LogColumnHeaders: React.FunctionComponent<{
       {columnConfigurations.map(columnConfiguration => {
         if (isTimestampLogColumnConfiguration(columnConfiguration)) {
           return (
-            <LogColumnHeader
-              columnWidth={columnWidths[columnConfiguration.timestampColumn.id]}
-              data-test-subj="logColumnHeader timestampLogColumnHeader"
-              key={columnConfiguration.timestampColumn.id}
-            >
-              Timestamp
-            </LogColumnHeader>
+            <WithLogPosition key={columnConfiguration.timestampColumn.id}>
+              {({ firstVisiblePosition }) => (
+                <LogColumnHeader
+                  columnWidth={columnWidths[columnConfiguration.timestampColumn.id]}
+                  data-test-subj="logColumnHeader timestampLogColumnHeader"
+                >
+                  {firstVisiblePosition ? localizedDate(firstVisiblePosition.time) : 'Timestamp'}
```

Comment:
👍 I like that `localizedDate` is a utility function which can be used directly, by hooks, etc